File: 36/python3/main.py
#!/usr/bin/env python3

class Solution:
    def isValidSudoku(self, board):
        for i in range(9):
            row_seen = set()
            col_seen = set()
            grid_seen = set()
            
            for j in range(9):
                # Check row duplicates
                if board[i][j] != '.':
                    if board[i][j] in row_seen:
                        return False
                    row_seen.add(board[i][j])

                # Check column duplicates
                if board[j][i] != '.':
                    if board[j][i] in col_seen:
                        return False
                    col_seen.add(board[j][i])

                # Check 3x3 grid duplicates
                grid_row = 3 * (i // 3) + j // 3
                grid_col = 3 * (i % 3) + j % 3
                if board[grid_row][grid_col] != '.':
                    if board[grid_row][grid_col] in grid_seen:
                        return False
                    grid_seen.add(board[grid_row][grid_col])
        
        return True

File: 36/cpp/main.cpp
#include <bitset>
#include <iostream>
#include <ranges>
#include <vector>

class Solution {
public:
    bool isValidSudoku(std::vector<std::vector<char>>& board) {
        // Check rows, columns, and 3x3 subgrids
        for (int i = 0; i < 9; ++i) {
            std::bitset<9> rowSeen, colSeen, gridSeen;
            for (int j = 0; j < 9; ++j) {
                // Check row duplicates
                if (board[i][j] >= '1' && board[i][j] <= '9') {
                    int rowBit = board[i][j] - '1';
                    if (rowSeen[rowBit]) return false;
                    rowSeen.set(rowBit);
                }

                // Check column duplicates
                if (board[j][i] >= '1' && board[j][i] <= '9') {
                    int colBit = board[j][i] - '1';
                    if (colSeen[colBit]) return false;
                    colSeen.set(colBit);
                }

                // Check 3x3 grid duplicates
                int gridRow = 3 * (i / 3) + j / 3;
                int gridCol = 3 * (i % 3) + j % 3;
                if (board[gridRow][gridCol] >= '1' && board[gridRow][gridCol] <= '9') {
                    int gridBit = board[gridRow][gridCol] - '1';
                    if (gridSeen[gridBit]) return false;
                    gridSeen.set(gridBit);
                }
            }
        }

        // No duplicates found, Sudoku is valid
        return true;
    }
};
